src/CommunityCommons/javasource/communitycommons/actions/GetImageDimensions.java

```diff
@@ -32,22 +35,39 @@ public GetImageDimensions(IContext context, IMendixObject ImageParameter)
 	@java.lang.Override
 	public IMendixObject executeAction() throws Exception
 	{
-		this.ImageParameter = __ImageParameter == null ? null : system.proxies.Image.initialize(getContext(), __ImageParameter);
+		this.ImageParameter = this.__ImageParameter == null ? null : system.proxies.Image.initialize(getContext(), __ImageParameter);
 
 		// BEGIN USER CODE
 		ImageDimensions imageDimensions = new ImageDimensions(getContext());
-		try (InputStream inputStream = Core.getImage(getContext(), this.ImageParameter.getMendixObject(), false)) {
-			BufferedImage bimg = ImageIO.read(inputStream);
-			imageDimensions.setHeight(bimg.getHeight());
-			imageDimensions.setWidth(bimg.getWidth());
-		}
 
-		return imageDimensions.getMendixObject();
+		try (InputStream inputStream = Core.getImage(getContext(), this.ImageParameter.getMendixObject(), false);
+			ImageInputStream imageStream = ImageIO.createImageInputStream(inputStream)) {
+			if(imageStream != null) {
+				Iterator<ImageReader> readers = ImageIO.getImageReaders(imageStream);
+				if (!readers.hasNext()) throw new IOException("Can't find a suitable imagereader for this image");;
+
+				ImageReader reader = readers.next();
+			  
+				try {
+					reader.setInput(imageStream);
+					int height = reader.getHeight(reader.getMinIndex());

```

Comment:
The PNG, JPG and BMP ImageReader implementations only read the imageheader to determine the height and width.